### PR TITLE
[11.0][MIG] base_action_rule -> base_automation

### PR DESCRIPTION
**Pull request by yvaucher**
Migration of base_action_rule into base_automation

Changes models, table and one XMLID

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: odoo/addons/openupgrade_records/lib/apriori.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 """
 
 renamed_modules = {
+    'base_action_rule': 'base_automation',
```
Comment (yvaucher) [inline odoo/addons/openupgrade_records/lib/apriori.py]: This will depends on ~~#1095~~ #1102

**Review by lepistone — APPROVED** [review]

File: addons/base_automation/migrations/11.0.1.0/openupgrade_analysis_work.txt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+base_automation / base.automation.lead.test / line_ids (one2many)           : NEW relation: base.automation.line.test
+base_automation / base.automation.lead.test / name (char)                   : NEW required: required
+base_automation / base.automation.lead.test / partner_id (many2one)         : NEW relation: res.partner
+base_automation / base.automation.lead.test / priority (boolean)            : NEW 
+base_automation / base.automation.lead.test / state (selection)             : NEW selection_keys: ['cancel', 'done', 'draft', 'open', 'pending']
+base_automation / base.automation.lead.test / user_id (many2one)            : NEW relation: res.users
+base_automation / base.automation.line.test / lead_id (many2one)            : NEW relation: base.automation.lead.test
+base_automation / base.automation.line.test / name (char)                   : NEW 
+base_automation / base.automation.line.test / user_id (many2one)            : NEW relation: res.users
+base_automation / ir.actions.server        / usage (False)                 : NEW selection_keys: ['base_automation', 'ir_actions_server', 'ir_cron'], mode: modify
```
Comment (StefanRijnhart) [inline addons/base_automation/migrations/11.0.1.0/openupgrade_analysis_work.txt:26]: Need to set usage to base_automation for actions that are linked to an action rule

File: addons/base_automation/migrations/11.0.1.0/pre-migration.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+from openupgradelib import openupgrade
+
+
+model_renames_base_action_rule = [
+    ('base.action.rule', 'base.automation')
+]
+
+table_renames_base_action_rule = [
+    ('base_action_rule', 'base_automation')
+]
```
Comment (yvaucher) [inline addons/base_automation/migrations/11.0.1.0/pre-migration.py:22]: @StefanRijnhart 
are model and table rename deprecated by apriori ?

https://github.com/OCA/OpenUpgrade/blob/11.0/odoo/addons/openupgrade_records/lib/apriori.py#L28

Comment (StefanRijnhart) [reply addons/base_automation/migrations/11.0.1.0/pre-migration.py:22]: No, we leave the renames to the originating modules. Apriori is to improve the analysis.

Comment (yvaucher) [reply addons/base_automation/migrations/11.0.1.0/pre-migration.py:22]: Ok

**Comment by pedrobaeza** [comment]
@yvaucher can you please finish this PR?

File: addons/base_automation/migrations/11.0.1.0/openupgrade_analysis_work.txt (hunk 2 above)
Comment (yvaucher) [reply addons/base_automation/migrations/11.0.1.0/openupgrade_analysis_work.txt:26]: Not sure what this usage field is, I'll check that tomorrow

**Comment by yvaucher** [comment]
@StefanRijnhart @pedrobaeza I made some advances on this.

I didn't expect that _inherits implied by the attribute `delegate` so I had to create `ir.actions.servers records`

Not sure if we want to drop the unused columns from table `base_automation` that I copy to `ir_act_server`.

And any idea of what to do with the 3 dropped fields? seems to me that we can't do much with them the methods of `base.automation` have been simplified.

File: addons/base_automation/migrations/11.0.1.0/openupgrade_analysis_work.txt (hunk 2 above)
Comment (yvaucher) [reply addons/base_automation/migrations/11.0.1.0/openupgrade_analysis_work.txt:26]: In fact we needs to create a ir.actions.server for each base.automation
The field server_action_id being set as delegate=True

File: addons/base_automation/migrations/11.0.1.0/pre-migration.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        'code': DEFAULT_PYTHON_CODE,
+        'state': 'object_write',
+        'type': 'ir.actions.server',
+        'usage': 'base_automation',
+    }
+    cr.execute("""
+        SELECT id, create_uid, create_date, write_uid, write_date
+          name, sequence, model_id, kind
+        FROM base_automation
+        WHERE action_server_id IS NULL
```
Comment (njeudy) [inline addons/base_automation/migrations/11.0.1.0/pre-migration.py]: Missing ";" ?

Comment (pedrobaeza) [reply addons/base_automation/migrations/11.0.1.0/pre-migration.py]: Not needed on pyscopg2 queries

**Comment by pedrobaeza** [comment]
I'm going to check pending things in this for finishing it.

**Comment by pedrobaeza** [comment]
> Not sure if we want to drop the unused columns from table base_automation that I copy to ir_act_server.

We should keep them for other migration scripts (even for custom modules) that make use of them for other purposes. Module database_cleanup can be used later for dropping the garbage (columns, fields, tables, models...)

> And any idea of what to do with the 3 dropped fields? seems to me that we can't do much with them the methods of base.automation have been simplified.

act_user_id is very specific and no special problem in not being transferred. If someone is using it for anything special, can take the SQL column. The other 2 serves for filtering records, and we should dump the domain in the filter to the domains fields (I have already done it).

**Review by pedrobaeza — APPROVED** [review]

**Comment by yvaucher** [comment]
@pedrobaeza thanks

**Comment by pedrobaeza** [comment]
@yvaucher there's more work in #1303

**Pull request merged**